packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx

```diff
@@ -102,48 +100,71 @@ const useStyles = makeStyles(({ spacing, typography, palette }) => ({
 }))
 
 type DrawerContentProps = {
-  data: ReturnType<typeof useDrawer<'InboxUnavailable'>>['data']
+  data: ReturnType<typeof useInboxUnavailableModal>['data']
+  onClose: () => void
 }
 
-const DrawerContent = ({ data }: DrawerContentProps) => {
+const DrawerContent = ({ data, onClose }: DrawerContentProps) => {
   const styles = useStyles()
   const dispatch = useDispatch()
   const navigation = useNavigation()
 
-  const { userId, shouldOpenChat } = data
-  const user = useSelector((state) => getUser(state, { id: userId }))
-  const { canCreateChat, callToAction } = useSelector((state) =>
+  const { userId, presetMessage } = data
+  const user = useSelector((state) =>
+    cacheUsersSelectors.getUser(state, { id: userId })
+  )
+  const { callToAction } = useSelector((state) =>
     getCanCreateChat(state, { userId })
   )
-
-  const closeDrawer = useCallback(() => {
-    dispatch(
-      setVisibility({
-        drawer: 'InboxUnavailable',
-        visible: false
-      })
-    )
-  }, [dispatch])
+  const currentUserId = useSelector(accountSelectors.getUserId)
 
   const handleUnblockPress = useCallback(() => {
-    dispatch(unblockUser({ userId }))
-    if (shouldOpenChat && canCreateChat) {
-      dispatch(createChat({ userIds: [userId] }))
+    if (!userId) {
+      console.error(
```

Comment:
Just double-checking this is sufficient handling of the error. Does the user see anything? Will we get a message in sentry or some other type of analytics. It reads as a silent failure as is.

Reply:
Yeah this is a good q. Sentry does capture our `console.error` messages, but I was torn about whether to throw here and crash the app, or just log an error, or do nothing. In theory, this should never happen - the drawer should always be opened with a userId - but typecheck makes some of these checks necessary and it felt like I should at least log them if they do occur.

Reply:
Makes sense. I just wanted to make sure that we have actionable data if this _does_ happen, as it would probably be in a context where something else has gone wrong in an unexpected way.

> but typecheck makes some of these checks necessary

This is a pattern worth thinking about. There's a way to design where values are _not_ typed as potentially null or undefined. That would essentially mean that we throw at a much lower layer (validating values as they come across the wire), and we don't even render a component that would consume a value in cases where the value doesn't exist.

It's kind of a fundamental shift, though. And not a trivial amount of effort, so I'm not pushing it 😅 . This is a reasonable approach given that we can't change the type to be required without doing a lot of other work.